examples/camera/projection_zoom.rs

```diff
@@ -0,0 +1,245 @@
+            ));
+            parent.spawn(TextBundle::from_section(
+                "A or D: yaw",
+                TextStyle::default(),
+            ));
+        });
+}
+
+fn orbit(
+    mut camera: Query<&mut Transform, With<Camera>>,
+    camera_settings: Res<CameraSettings>,
+    keyboard_input: Res<ButtonInput<KeyCode>>,
+    time: Res<Time>,
+) {
+    let mut transform = camera.single_mut();
+
+    let mut delta_pitch = 0.0;
+    let mut delta_yaw = 0.0;
+
+    if keyboard_input.pressed(KeyCode::KeyW) {
+        delta_pitch += camera_settings.orbit_speed;
+    }
+    if keyboard_input.pressed(KeyCode::KeyA) {
+        delta_yaw -= camera_settings.orbit_speed;
+    }
+    if keyboard_input.pressed(KeyCode::KeyS) {
+        delta_pitch -= camera_settings.orbit_speed;
+    }
+    if keyboard_input.pressed(KeyCode::KeyD) {
+        delta_yaw += camera_settings.orbit_speed;
+    }
+
+    // Incorporating the delta time between calls prevents this from being framerate-bound.
+    delta_pitch *= time.delta_seconds();
+    delta_yaw *= time.delta_seconds();
+
+    // Obtain the existing pitch, yaw, and roll values from the transform.
+    let (yaw, pitch, roll) = transform.rotation.to_euler(EulerRot::YXZ);
+
+    // Establish the new yaw and pitch, preventing the pitch value from exceeding our limits.
+    let pitch = (pitch + delta_pitch).clamp(
+        camera_settings.pitch_range.start,
+        camera_settings.pitch_range.end,
+    );
+    let yaw = yaw + delta_yaw;
+    transform.rotation = Quat::from_euler(EulerRot::YXZ, yaw, pitch, roll);
+
+    // Adjust the translation to maintain the correct orientation toward the orbit target.
+    transform.translation = Vec3::ZERO - transform.forward() * camera_settings.orbit_distance;
+}
+
+fn switch_projection(
+    mut camera: Query<&mut Projection, With<Camera>>,
+    camera_settings: Res<CameraSettings>,
+    keyboard_input: Res<ButtonInput<KeyCode>>,
+) {
+    let mut projection = camera.single_mut();
+
+    if keyboard_input.just_pressed(KeyCode::Space) {
+        // Switch projection type
+        *projection = match *projection {
+            Projection::Orthographic(_) => Projection::Perspective(PerspectiveProjection {
+                fov: camera_settings.perspective_zoom_range.start,
+                ..default()
+            }),
+            Projection::Perspective(_) => Projection::Orthographic(OrthographicProjection {
+                scaling_mode: ScalingMode::FixedVertical(
+                    camera_settings.orthographic_zoom_range.start,
+                ),
+                ..default()
+            }),
+        }
+    }
+}
+
+fn zoom(
+    mut camera: Query<&mut Projection, With<Camera>>,
+    camera_settings: Res<CameraSettings>,
+    mouse_wheel_input: Res<AccumulatedMouseScroll>,
+) {
+    let mut projection = camera.single_mut();
+
+    // Usually, you won't need to handle both types of projection. This is by way of demonstration.
+    match &mut *projection {
```

Comment:
```suggestion
    match projection.deref_mut() {
```

Reply:
OMG THANK YOU! That was driving me nuts.